File: commands/Fun/numberfact.js
const { Command } = require("klasa");
const { get } = require("superagent");

class NumberFact extends Command {
  constructor(...args) {
    super(...args, {
      description: "Get a fact about a number or random number",
      usage: "(number:numfact)",
      cooldown: 5,
      aliases: ["numfact", "numfacts", "num", "number", "number-fact", "number-facts"]
    });

    this.createCustomResolver("numfact", (arg, possible, msg) => {
      if(!arg || arg.toLowerCase() === "random") return "random";
      return this.client.arguments.get("integer").run(arg, possible, msg);
    });
  }

  async run(msg, [number]) {
    const { text } = await get(`http://numbersapi.com/${number}`);
    return msg.send(`**${text}**`);
  }
}

module.exports = NumberFact;

File: commands/fun/numberfact.go
package fun

import (
  "net/http"
  "github.com/sapphire-cord/sapphire"
  "io/ioutil"
  "strconv"
)

// Get a fact about a number.
// Usage: [number:int]
// Aliases: numfact, numfacts, numbefacts
func Numberfact(ctx *sapphire.CommandContext) {
  num := "random"

  if ctx.HasArgs() {
    num = strconv.Itoa(ctx.Arg(0).AsInt())
  }

  res, err := http.Get("http://numbersapi.com/" + num)
  if err != nil {
    ctx.Reply("%s", err)
    return
  }
  defer res.Body.Close()
  body, err := ioutil.ReadAll(res.Body)
  if err != nil {
    ctx.Reply("%s", err)
    return
  }
  ctx.Reply("**%s**", string(body))
}
